extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -424,4 +466,71 @@ export class CreateProjectFromDatabaseDialog {
 			throw new Error(constants.extractTargetRequired);
 		}
 	}
+
+	async validate(): Promise<boolean> {
+		try {
+			// the selected location should be an existing directory
+			const parentDirectoryExists = await exists(this.projectLocationTextBox!.value!);
+			if (!parentDirectoryExists) {
+				this.showErrorMessage(constants.ProjectParentDirectoryNotExistError(this.projectLocationTextBox!.value!));
+				return false;
+			}
+
+			// there shouldn't be an existing sub directory with the same name as the project in the selected location
+			const projectDirectoryExists = await exists(path.join(this.projectLocationTextBox!.value!, this.projectNameTextBox!.value!));
+			if (projectDirectoryExists) {
+				this.showErrorMessage(constants.ProjectDirectoryAlreadyExistError(this.projectNameTextBox!.value!, this.projectLocationTextBox!.value!));
+				return false;
+			}
+
+			if (this.workspaceInputBox!.enabled && !await this.validateNewWorkspace()) {
+				return false;
+			}
+
+			return true;
+		} catch (err) {
+			this.showErrorMessage(err?.message ? err.message : err);
+			return false;
+		}
+	}
+
+	protected async validateNewWorkspace(): Promise<boolean> {
+		const sameFolderAsNewProject = path.join(this.projectLocationTextBox!.value!, this.projectNameTextBox!.value!) === path.dirname(this.workspaceInputBox!.value!);
+
+		// workspace file should end in .code-workspace
+		const workspaceValid = this.workspaceInputBox!.value!.endsWith(constants.WorkspaceFileExtension);
+		if (!workspaceValid) {
```

Comment:
Same comment here: may as well just throw errors and let them bubble all the way up until they need to be caught/handled.